applications/neuroscan/frontend/src/components/viewers/viewer.js

```diff
@@ -95,20 +96,38 @@ class Viewer extends React.Component {
 
     this.timeoutRef = React.createRef();
     this.tooltipRef = React.createRef();
-
     this.onMount = this.onMount.bind(this);
     this.onSelection = this.onSelection.bind(this);
     this.hoverListener = this.hoverListener.bind(this);
     this.initCanvasData = this.initCanvasData.bind(this);
   }
 
+  componentDidMount() {
+    this.handleDeleteKeyPress = (event) => {
+      if ((event.key === 'Delete' || event.key === 'Backspace')) {
+        const { selectedInstanceToDelete, widgets } = this.props;
+        const currentWidget = widgets[selectedInstanceToDelete.viewerId];
+        const { config } = currentWidget;
```

Comment:
@Salam-Dalloul suggestion: move this logic (getting the config of the viewer of the last selected instance) to the deleteSelectedInstances function. 
And then call the deleteSelectedInstances function with the viewerId and instance (I think it's the UID and not the instance, if so then please change the name of the value in the selectedInstance object)